### PR TITLE
Feature/aa 138 discount management

### PR DESCRIPTION
**Pull request by mingazhev**

### CODE REVIEW

File: Api/AdministratorServices/AgencyDiscountManagementService.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    {
+        public AgencyDiscountManagementService(EdoContext context)
+        {
+            _context = context;
+        }
+
+
+        public Task<List<DiscountInfo>> Get(int agencyId)
+        {
+            return (from discount in _context.Discounts
```
Comment (kirillta) [inline Api/AdministratorServices/AgencyDiscountManagementService.cs:24]: Database queries must materialize at place according to our rules

File: Api/AdministratorServices/AgencyDiscountManagementService.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                    Description = discount.Description,
+                    TargetMarkupId = discount.TargetPolicyId,
+                    TargetPolicyDescription = markupPolicy.Description,
+                    IsActive = discount.IsActive
+                }).ToListAsync();
+        }
+
+
+        public async Task<Result> Activate(int agencyId, int discountId)
+        {
```
Comment (kirillta) [inline Api/AdministratorServices/AgencyDiscountManagementService.cs:38]: Could be one-lined

File: Api/AdministratorServices/AgencyDiscountManagementService.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            return await Get(agencyId, discountId)
+                .Map(d => Update(d, discount => discount.IsActive = true));
+        }
+
+
+        public async Task<Result> Deactivate(int agencyId, int discountId)
+        {
```
Comment (kirillta) [inline Api/AdministratorServices/AgencyDiscountManagementService.cs:45]: Could be one-lined

File: Api/AdministratorServices/AgencyDiscountManagementService.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                    return Result.Failure("Cannot apply discount to non-global markup policy");
+
+                return Result.Success();
+            }
+
+
+            async Task<Result> ValidateAgency()
+                => await _context.Agencies.AnyAsync(a => a.Id == agencyId)
+                    ? Result.Success()
+                    : Result.Failure($"Cannot add an agency with id {agencyId}");
```
Comment (kirillta) [inline Api/AdministratorServices/AgencyDiscountManagementService.cs:77]: `Cannot get` or `Cannot find` I presume

File: Api/AdministratorServices/IAgencyDiscountManagementService.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace HappyTravel.Edo.Api.AdministratorServices
+{
+    public interface IAgencyDiscountManagementService
+    {
+        Task<List<DiscountInfo>> Get(int agencyId);
+        
+        Task<Result> Activate(int agencyId, int discountId);
+        
+        Task<Result> Deactivate(int agencyId, int discountId);
```
Comment (kirillta) [inline Api/AdministratorServices/IAgencyDiscountManagementService.cs:14]: `Start` and `Stop` are more suitable here

**Comment by kirillta** [comment]
@DFSko could you finish this PR, please?

File: Api/AdministratorServices/AgencyDiscountManagementService.cs (hunk 2 above)
Comment (DFSko) [reply Api/AdministratorServices/AgencyDiscountManagementService.cs]: Fixed

File: Api/AdministratorServices/AgencyDiscountManagementService.cs (hunk 3 above)
Comment (DFSko) [reply Api/AdministratorServices/AgencyDiscountManagementService.cs]: Fixed

File: Api/AdministratorServices/AgencyDiscountManagementService.cs (hunk 4 above)
Comment (DFSko) [reply Api/AdministratorServices/AgencyDiscountManagementService.cs]: Fixed

File: Api/AdministratorServices/IAgencyDiscountManagementService.cs (hunk 5 above)
Comment (DFSko) [reply Api/AdministratorServices/IAgencyDiscountManagementService.cs]: Fixed

File: Api/AdministratorServices/AgencyDiscountManagementService.cs (hunk 1 above)
Comment (DFSko) [reply Api/AdministratorServices/AgencyDiscountManagementService.cs]: Fixed

File: Api/Controllers/AdministratorControllers/AgencyDiscountsController.cs [hunk 6]
```diff
@@ -69,7 +69,7 @@ public async Task<IActionResult> AddDiscount([FromRoute] int agencyId, [FromBody
         [AdministratorPermissions(AdministratorPermissions.MarkupManagement)]
         public async Task<IActionResult> DeactivateDiscount([FromRoute] int agencyId, [FromRoute] int discountId)
```
Comment (kirillta) [inline Api/Controllers/AdministratorControllers/AgencyDiscountsController.cs:70]: Change the method names as well

**Review by kirillta — APPROVED** [review]

**Pull request merged**